### ISSUE TITLE
408 authentication time out on new install

### ISSUE THREAD

**Issue by E-dk** [bigbluebutton/bbb-install#284]
After a new install i have the error 'Error while logging in' Authentication Time Out 408. when i connect from Moodle. I have another install thats working fine which i installed a few months ago with the same installation options. The only slightly resembling issue i could find was [https://github.com/bigbluebutton/bigbluebutton/issues/9400](url). Because this server also has a lot of cores (96). The fixes proposed there don't work for me. Any suggestions?

**Comment by ffdixon** [comment]
Are you able to install the API examples and launch into a meeting?  We just tried using 2.2.28 (soon to be released) with Moodle and didn't get any errors on startup.

If you want to try 2.2.28 on your server, run `bbb-install.sh` and use the parameter `-v  xenial-220-dev22a`.

Also, 96 cores is overkill at the moment.  The nodejs process within mongodb runs on a single CPU core, so in 2.2 it constrains how many users a BigBlueButton server can support.  We're working on improving this in 2.3.

**Comment by E-dk** [comment]
Fred, thanks for your reaction. I will try 2.2.28. Actually i tried 2.3 last evening but that didn't solve the problem. Regarding to the number of cores, maybe a workaround could be to install hyper-v on this server and install BBB as one or two HV guest(s)? Or is this still not advisable because of Freeswitch requirements? I have a environment with 1500 students and 100 teachers, due to covid19 i assume at some point i might have 500 concurrent users in the near feature.

**Comment by E-dk** [comment]
Tried 2.2.28 to no avail. My gut feeling says problem is ubuntu-hardware related. New Poweredge r6525 with 2 AMD EPYC 7642 procs.

**Comment by E-dk** [comment]
Installed as a HV guest on the same server (now with windows server 2019 installed), assigned 16 cores, works instantly. Just fyi.

**Comment by ffdixon** [comment]
Thanks for sharing -- it would be better to use a number of smaller VMs (with 8 CPU cores/16G of memory) front-ended by [scalelite](https://github.com/blindsidenetworks/scalelite).

**Comment by E-dk** [comment]
Thanks again for your input. I'm going to try scalelite - until i have enough knowledge i will split the population in 3 or 4 portals with corresponding bbb severs.

**Comment by paul1278** [comment]
I had the same problem on v2.2.28 and digged into the race condition between `bbb-apps-akka` and `redis` as mentioned here https://github.com/bigbluebutton/bigbluebutton/issues/9400#issuecomment-649523930
After analyzing the systemd-startup tree using `systemd-analyze plot > tree.svg` I saw that `bbb-apps-akka` starts ~4s before redis.

For me the solution was to create an override-file for `redis-server` using `systemctl edit redis-server.service` with the content:

```
[Unit]
Before=bbb-apps-akka.service
```

Now `redis` starts before `bbb-apps-akka`, I will report new issues triggered by this solution here, but for now that works.

**Comment by Thatoo** [comment]
I face the same issue with the same situation. I'm trying @paul1278 fix and we'll let you know if it helps.

**Comment by tcg71** [comment]
Same problem here after first update on a new system created 3 weeks ago: The solution @paul1278 provided is working, but still needs a `bbb-conf --restart` after each system restart.

**Comment by paul1278** [comment]
@tcg71 it works without a restart here, maybe there is another race condition somewhere, you could try analyzing your startup using the command mentioned earlier: `systemd-analyze plot > tree.svg` and then open the tree.svg-file.

**Comment by tcg71** [comment]
@paul1278 I created the file, but I can only see that bbb_record_core.slice starts before the network is ready. But I don't see any hint of redis and akka. 

[tree.svg.zip](https://github.com/bigbluebutton/bbb-install/files/5579363/tree.svg.zip)

**Comment by paul1278** [comment]
@tcg71 do you start redis and akka in Docker-containers? Because then you need to put those in correct order

**Comment by crosscodr** [comment]
I also couldn't see much in my tree.svg from `systemd-analyze`, and found this hint from https://www.freedesktop.org/software/systemd/man/systemd-analyze.html:
> Also note: systemd-analyze blame doesn't display results for services with Type=simple, because systemd considers such services to be started immediately, hence no measurement of the initialization delays can be done.

Maybe that's a problem, so that systemd can't prevent race conditions in all cases.
Found these bbb-services to be of type simple:
```
bbb-apps-akka.service:Type=simple
bbb-fsesl-akka.service:Type=simple
bbb-transcode-akka.service:Type=simple
bbb-web.service:Type=simple
 ```

**Comment by ffdixon** [comment]
We're adding some dependencies in 2.2.30 to ensure redis starts first.  Also added to bbb-install.sh if you use it now to setup BigBlueButton.

  https://github.com/bigbluebutton/bbb-install/blob/master/bbb-install.sh#L363

**Comment by E-dk** [comment]
Fred i assume its still better to split the load over multiple virtual bbb servers with scalelite, than one large physical bbb server. I now have 8 virtual servers on 2 physical servers with scalelite in front. Works excellent.

**Comment by paul1278** [comment]
@E-dk yes it still is, my fix is not for that problem, but for the problem on starting BigBlueButton on bigger servers.

**Comment by mahdiyardiyanat** [comment]
Hello to all 

About 2 months after installing the new version of BigBlueButton on my dedicated server in Ubuntu 1804, I had no problem until today, but from today I encountered "app.error.408".
I followed your instructions in GateHub, but I could not solve my problem
please help me
 
New release '20.04.3 LTS' available.
Run 'do-release-upgrade' to upgrade to it.
 
Last login: Sun Feb  6 00:42:08 2022 from 46.167.141.56
root@server13:~# bbb-conf --check
 
BigBlueButton Server 2.3.17 (2525)
                    Kernel version: 4.15.0-163-generic
                      Distribution: Ubuntu 18.04.6 LTS (64-bit)
                            Memory: 9986 MB
                         CPU cores: 12
 
/etc/bigbluebutton/bbb-web.properties (override for bbb-web)
/usr/share/bbb-web/WEB-INF/classes/bigbluebutton.properties (bbb-web)
       bigbluebutton.web.serverURL: https://server13.lmsschool.ir
                defaultGuestPolicy: ALWAYS_ACCEPT
                 svgImagesRequired: true
 
/etc/nginx/sites-available/bigbluebutton (nginx)
                       server_name: server13.lmsschool.ir
                              port: 80, [::]:80
                              port: 443 ssl
 
/opt/freeswitch/etc/freeswitch/vars.xml (FreeSWITCH)
                       local_ip_v4: 91.98.31.166
                   external_rtp_ip: 91.98.31.166
                   external_sip_ip: 91.98.31.166
 
/opt/freeswitch/etc/freeswitch/sip_profiles/external.xml (FreeSWITCH)
                        ext-rtp-ip: $${local_ip_v4}
                        ext-sip-ip: $${local_ip_v4}
                        ws-binding: 91.98.31.166:5066
                       wss-binding: 91.98.31.166:7443
 
/usr/local/bigbluebutton/core/scripts/bigbluebutton.yml (record and playback)
                     playback_host: server13.lmsschool.ir
                 playback_protocol: https
                            ffmpeg: 4.2.4-1ubuntu0.1bbb2~18.04
 
/etc/bigbluebutton/nginx/sip.nginx (sip.nginx)
                        proxy_pass: 91.98.31.166
                          protocol: http
 
/usr/local/bigbluebutton/bbb-webrtc-sfu/config/default.yml (Kurento SFU)
                        kurento.ip: 91.98.31.166
                       kurento.url: ws://127.0.0.1:8888/kurento
                    kurento.sip_ip: 91.98.31.166
                    localIpAddress: 91.98.31.166
               recordScreenSharing: true
                     recordWebcams: true
                  codec_video_main: VP8
               codec_video_content: VP8
 
/usr/share/meteor/bundle/programs/server/assets/app/config/settings.yml (HTML5 client)
/etc/bigbluebutton/bbb-html5.yml (HTML5 client config override)
                             build: 1865
                        kurentoUrl: wss://server13.lmsschool.ir/bbb-webrtc-sfu
                  enableListenOnly: true
                    sipjsHackViaWs: true
 
/usr/share/bbb-web/WEB-INF/classes/spring/turn-stun-servers.xml (STUN Server)
                              stun: stun.l.google.com:19302
 
 
# Potential problems described below
# Not running:  Nginx tomcat8 or grails
 
#
# Error: Unable to connect to the FreeSWITCH Event Socket Layer on port 8021
#
curl: (7) Failed to connect to server13.lmsschool.ir port 443: Connection refused
.curl: (7) Failed to connect to server13.lmsschool.ir port 443: Connection refused
.curl: (7) Failed to connect to server13.lmsschool.ir port 443: Connection refused
.^C
root@server13:~#
-----------------------------------------------------------
root@server13:~# bbb-conf --status
nginx —————————————————► [✘ - inactive]
freeswitch ————————————► [✘ - failed]
redis-server ——————————► [✘ - deactivating]
bbb-apps-akka —————————► [✘ - failed]
bbb-fsesl-akka ————————► [✘ - failed]
tomcat8 ———————————————► [✘ - inactive]
mongod ————————————————► [✘ - inactive]
bbb-html5 —————————————► [✘ - inactive]
bbb-webrtc-sfu ————————► [✘ - inactive]
kurento-media-server ——► [✘ - inactive]
bbb-html5-backend@1 ———► [✘ - inactive]
bbb-html5-backend@2 ———► [✘ - inactive]
bbb-html5-frontend@1 ——► [✘ - inactive]
bbb-html5-frontend@2 ——► [✘ - inactive]
etherpad ——————————————► [✘ - inactive]
bbb-web ———————————————► [✘ - failed]
root@server13:~#

I realized that the above should be active but I do not know how to do it

**Comment by ffdixon** [comment]
Is it an option for you to install the latest version of 2.4?  It will be easier for us to help if you work with the current released version.

**Comment by mahdiyardiyanat** [comment]
Yes, I installed the latest version of Big Blue Button 2.4 and my Ubuntu version is 1804   Sent from Mail for Windows From: Fred DixonSent: ‏يكشنبه,‏ ‏17 ‏بهمن ‏1400 ‏06:35 ‏ق.ظTo: bigbluebutton/bbb-installCc: mahdiyardiyanat; CommentSubject: Re: [bigbluebutton/bbb-install] 408 authentication time out on new install (#284) Is it an option for you to install the latest version of 2.4? It will be easier for us to help if you work with the current released version.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

**Comment by mahdiyardiyanat** [comment]
@ffdixon 

Hello Dear
I try to make my problem clearer
For now, I want to ignore the problem I mentioned and address another issue that may be related to this one.
I turned off the BigBlueButton server that and I had installed and had a problem and re-installed the latest version of BigBlueButton and Greenlight on Ubuntu version 1804
I went through all the settings and installation steps correctly and the rooms that are being built in BigBlueButton are also working properly so far.
My problem starts when I create a room on my site using secret code BigBlueButton, but I can not access it. I enter the address "https://mconf.github.io/api-mate/#server=https: //server14.lmsschool.ir/bigbl "and matched the room ID made in BigBlueButton with the rooms made in the site and the IDs was  the same, but it is not possible for me to enter the room through the site and With the message "<response>
<returncode> FAILED </returncode>
<messageKey> checksumError </messageKey>
<message> Checksums do not match </message>
</response> " thank you for your help. It should be noted that my site is working properly on another server, but I have a problem with this server
Tell me if you need other information. thank you
